### ISSUE TITLE
[CRASH]  opensips-3.1.2 segfault

### ISSUE THREAD

**Issue by xiaoluweiba** [OpenSIPS/opensips#2622]
**OpenSIPS version you are running**

version: opensips 3.1.2 (x86_64/linux)
flags: STATS: On, DISABLE_NAGLE, USE_MCAST, SHM_MMAP, PKG_MALLOC, Q_MALLOC, FAST_LOCK-ADAPTIVE_WAIT
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144, MAX_LISTEN 16, MAX_URI_SIZE 1024, BUF_SIZE 65535
poll method support: poll, epoll, sigio_rt, select.
main.c compiled on 22:08:05 Aug 30 2021 with gcc 9

**openssl version**: OpenSSL 1.1.1f  31 Mar 2020
```
```

**Crash Core Dump**

![image](https://user-images.githubusercontent.com/14882870/132163423-4b867fb3-aae8-440a-ab78-c6ea94647ede.png)

**Describe the traffic that generated the bug**

step1. load_module "proto_tls.so"
step2.  db_url set connec mysql  (**mysql  Ver 8.0.25-cluster for Linux on x86_64 (MySQL Cluster Community Server - GPL**))
step3. run opensips with listen 5 ports, it will crash by sometimes, abount 15%~20%

**To Reproduce**

**Relevant System Logs**

**OS/environment information**
 - Operating System: 
 - OpenSIPS installation: 
 - other relevant information:

**Additional context**

**Comment by xiaoluweiba** [comment]
and I have a solution to avoid the problem

**unsigned int ssl_mode = SSL_MODE_DISABLED;
mysql_options(ptr->con, MYSQL_OPT_SSL_MODE, (void *)&ssl_mode);**

before **mysql_real_connect**

**Comment by liviuchircu** [comment]
Thanks for the hints, @xiaoluweiba, I think I have a good lead as to what's happening here.

Basically, in early 2017, MySQL seem to have backported a new set of SSL-related settings in all supported **minor releases** named `MYSQL_OPT_SSL_MODE`, as follows:

* starting with `5.5.55`, [default value](https://downloads.mysql.com/docs/mysql-5.5-relnotes-en.a4.pdf) is `OFF`
* starting with `5.6.36`, [default value](https://dev.mysql.com/doc/refman/5.6/en/connection-options.html#option_general_ssl-mode) seems to be `OFF`, it's not very well documented.
* starting with `5.7.11`, [default value](https://dev.mysql.com/doc/c-api/5.7/en/mysql-options.html) is `ON`

While I don't think their backport necessarily changed the behavior across **minor** releases, it seems that the `5.7` client library has enabled encrypted connections by default.  Looking at MariaDB, they don't seem to have made this step forward.

So, basically, we should just force the `MYSQL_OPT_SSL_MODE` to be synchronized with the desired module parameter setting of [use_tls](https://opensips.org/docs/modules/3.3.x/db_mysql.html#param_use_tls).   If the script developer does not enable TLS for their connection, it should just stay OFF.

Does this clear your "problems" concern?

**Comment by xiaoluweiba** [comment]
Great！although my english is not good, but I fully understand what you are saying
